kubernetes_platform/proto/kubernetes_executor_config.proto

```diff
@@ -176,3 +178,23 @@ message Toleration {
     string effect = 4;
     optional int64 toleration_seconds = 5;
 }
+
+message NodeAffinity{
+    string key = 1;
+    string operator = 2;
+    string value = 3;
+    //Setting the weight makes it use PreferredDuringSchedulingIgnoredDuringExecution type of affinities instead of RequiredDuringSchedulingIgnoredDuringExecution type of affinities

```

Comment:
Just to confirm, so by default we are using `RequiredDuringSchedulingIgnoredDuringExecution` if weight is not defined?

Reply:
That's my intention indeed. if we do not set weight, weight default to a value of 0.  (See https://github.com/cjidboon94/pipelines/blob/c902efcef860b2e96a79693fa2336c18b66ae73c/kubernetes_platform/go/kubernetesplatform/kubernetes_executor_config.pb.go#L1198-L1203 )

However, `weight` field (which is only used in the `PreferredDuringSchedulingIgnoredDuringExecution` Node/PodAffinity) is required to be a value between 1 and 100. So, if weight is equal to 0 , that should indicate to the driver to set a `RequiredDuringSchedulingIgnoredDuringExecution`

I will add the necessary transforms in the driver.go PR and checks in the python PR, when this one is approved.